### PR TITLE
update NCCL to 2.19.3

### PR DESCRIPTION
**Pull request by ptrblck**
CC @atalman @malfet 

Should fix: https://github.com/pytorch/pytorch/issues/116977

### CODE REVIEW

**Review by malfet — APPROVED** [review]
This is really weird, I though I've made this change in the past, but guess I did not :(

P.S. It uses nccl for cuda-12.0 vs 12.1, is this OK?

**Comment by atalman** [comment]
@malfet Yes looks like we would need to use 12.0 cuda nccl : https://github.com/NVIDIA/nccl/issues/1093

File: common/install_cuda.sh [hunk 1]
```diff
@@ -67,10 +67,10 @@ function install_121 {
 
     # NCCL license: https://docs.nvidia.com/deeplearning/nccl/#licenses
     mkdir tmp_nccl && cd tmp_nccl
-    wget -q https://developer.download.nvidia.com/compute/redist/nccl/v2.18.1/nccl_2.18.1-1+cuda12.1_x86_64.txz
-    tar xf nccl_2.18.1-1+cuda12.1_x86_64.txz
-    cp -a nccl_2.18.1-1+cuda12.1_x86_64/include/* /usr/local/cuda/include/
-    cp -a nccl_2.18.1-1+cuda12.1_x86_64/lib/* /usr/local/cuda/lib64/
+    wget -q https://developer.nvidia.com/downloads/compute/machine-learning/nccl/secure/2.19.3/agnostic/x64/nccl_2.19.3-1+cuda12.0_x86_64.txz
```
Comment (atalman) [inline common/install_cuda.sh:70]:
```suggestion
    wget -q https://ossci-linux.s3.amazonaws.com/nccl_2.19.3-1%2Bcuda12.0_x86_64.txz
```

Comment (atalman) [reply common/install_cuda.sh:70]: The secure URL may not be available. Hence lets use same version uploaded to pytorch s3 path.

**Review by huydhn — APPROVED** [review]

**Comment by bryantbiggs** [comment]
superseded by https://github.com/pytorch/builder/pull/1724

**Comment by ptrblck** [comment]
Closing

**Pull request closed without merging**